internal/pkg/update/github.go

```diff
@@ -0,0 +1,231 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package update
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v35/github"
+	"golang.org/x/oauth2"
+)
+
+func gitHubTokenFromEnv() string {
+	token := os.Getenv("BLDR_GITHUB_TOKEN")
+	if token == "" {
+		token = os.Getenv("GITHUB_TOKEN")
+	}
+	return token
+}
+
+type gitHub struct {
+	c *github.Client
+}
+
+func newGitHub(token string) *gitHub {
+	var c *http.Client
+	if token != "" {
```

Comment:
It feels a bit strange to return a client without a token. Will this work without one? I guess it will, just will get hit with rate limit.

Reply:
Exactly. 60 requests per hour are allowed for unauthenticated users. That may be enough for casual usage, but in the case of the rate limit, the error message mentions environment variables that could be set to increase it.